flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -18,130 +18,88 @@
 import org.apache.flink.runtime.query.TaskKvStateRegistry;
-import org.apache.flink.runtime.query.netty.DisabledKvStateRequestStats;
 import org.apache.flink.runtime.query.netty.KvStateServer;
-import org.apache.flink.runtime.taskmanager.NetworkEnvironmentConfiguration;
 import org.apache.flink.runtime.taskmanager.Task;
 import org.apache.flink.runtime.taskmanager.TaskManager;
-import org.apache.flink.util.Preconditions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import scala.Option;
 import scala.Tuple2;
-import scala.concurrent.ExecutionContext;
-import scala.concurrent.Future;
-import scala.concurrent.duration.FiniteDuration;
 
 import java.io.IOException;
 
-import static org.apache.flink.runtime.messages.JobManagerMessages.ScheduleOrUpdateConsumers;
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
 /**
  * Network I/O components of each {@link TaskManager} instance. The network environment contains
  * the data structures that keep track of all intermediate results and all data exchanges.
- *
- * When initialized, the NetworkEnvironment will allocate the network buffer pool.
- * All other components (netty, intermediate result managers, ...) are only created once the
- * environment is "associated" with a TaskManager and JobManager. This happens as soon as the
- * TaskManager actor gets created and registers itself at the JobManager.
  */
 public class NetworkEnvironment {
 
 	private static final Logger LOG = LoggerFactory.getLogger(NetworkEnvironment.class);
 
 	private final Object lock = new Object();
 
-	private final NetworkEnvironmentConfiguration configuration;
-
-	private final FiniteDuration jobManagerTimeout;
-
 	private final NetworkBufferPool networkBufferPool;
 
-	private ConnectionManager connectionManager;
+	private final ConnectionManager connectionManager;
 
-	private ResultPartitionManager partitionManager;
+	private final ResultPartitionManager resultPartitionManager;
 
-	private TaskEventDispatcher taskEventDispatcher;
-
-	private ResultPartitionConsumableNotifier partitionConsumableNotifier;
-
-	private PartitionStateChecker partitionStateChecker;
+	private final TaskEventDispatcher taskEventDispatcher;
 
 	/** Server for {@link org.apache.flink.runtime.state.KvState} requests. */
-	private KvStateServer kvStateServer;
+	private final KvStateServer kvStateServer;
 
 	/** Registry for {@link org.apache.flink.runtime.state.KvState} instances. */
-	private KvStateRegistry kvStateRegistry;
+	private final KvStateRegistry kvStateRegistry;
 
-	private boolean isShutdown;
+	private final IOMode defaultIOMode;
 
-	/**
-	 * ExecutionEnvironment which is used to execute remote calls with the
-	 * {@link JobManagerResultPartitionConsumableNotifier}
-	 */
-	private final ExecutionContext executionContext;
+	private final Tuple2<Integer, Integer> partitionRequestInitialAndMaxBackoff;
 
-	private final InstanceConnectionInfo connectionInfo;
+	private boolean isShutdown;
```

Comment:
That's a good point. I'll actually synchronize on `lock` because all other accesses to `isShutdown` happen also under `lock`. This eliminate the case where `isShutdown` returns false but some of the `NetworkEnvironment's` components are already shut down.